blas/tpls/KokkosBlas2_gemv_tpl_spec_decl.hpp

```diff
@@ -786,4 +786,105 @@ KOKKOSBLAS2_CGEMV_ROCBLAS(Kokkos::LayoutRight, Kokkos::Experimental::HIPSpace,
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<const SCALAR*, LAYOUT,                                      \
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<SCALAR*, LAYOUT,                                            \
+                   Kokkos::Device<Kokkos::Experimental::SYCL, MEM_SPACE>,      \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      true, ETI_SPEC_AVAIL> {                                                  \
+    using execution_space = Kokkos::Experimental::SYCL;                        \
+    using device_type     = Kokkos::Device<execution_space, MEM_SPACE>;        \
+    using mem_traits      = Kokkos::MemoryTraits<Kokkos::Unmanaged>;           \
+    using AViewType =                                                          \
+        Kokkos::View<const SCALAR**, LAYOUT, device_type, mem_traits>;         \
+    using XViewType =                                                          \
+        Kokkos::View<const SCALAR*, LAYOUT, device_type, mem_traits>;          \
+    using YViewType = Kokkos::View<SCALAR*, LAYOUT, device_type, mem_traits>;  \
+                                                                               \
+    static void gemv(const typename AViewType::execution_space& space,         \
+                     const char kk_trans[],                                    \
+                     typename AViewType::const_value_type& alpha,              \
+                     const AViewType& A, const XViewType& X,                   \
+                     typename YViewType::const_value_type& beta,               \
+                     const YViewType& Y) {                                     \
+      execution_space exec{};                                                  \
+      bool row_major       = std::is_same<Kokkos::LayoutRight, LAYOUT>::value; \
+      const std::int64_t M = A.extent(0);                                      \
+      const std::int64_t N = A.extent(1);                                      \
+      oneapi::mkl::transpose trans = mode_kk_to_onemkl(kk_trans[0]);           \
+      const std::int64_t LDA       = row_major ? A.stride(0) : A.stride(1);    \
+      std::string label            = "KokkosBlas::gemv[TPL_ONEMKL," +          \
+                          Kokkos::ArithTraits<STDSCALAR>::name() + "]";        \
+      sycl::event event;                                                       \
+                                                                               \
+      Kokkos::Profiling::pushRegion(label);                                    \
+      const STDSCALAR* a = reinterpret_cast<const STDSCALAR*>(A.data());       \
```

Comment:
This is okay, but I would have looked into a compile time check like:
```
if constexpr (Kokkos::ArithTraits<SCALAR>::is_complex) {
const std::complex<Kokkos::ArithTraits<SCALAR>::mag_type>* a = reinterpret_cast<...>(A.data());
} else {
 // no reinterpret cast, just use the pointers directly
}
```

Reply:
Tried but found this problem: `a` is undefined in "CODE USING a"

```
if constexpr (Kokkos::ArithTraits<SCALAR>::is_complex) {
   const std::complex<Kokkos::ArithTraits<SCALAR>::mag_type>* a = reinterpret_cast<...>(A.data());
} else {
  const SCALAR *a = A.data();
}

CODE USING a;
```
I have to duplicate "CODE USING a" to the if-else clauses to solve the problem.  Thus I had another approach using `kokkos_to_std_type_map`